### ISSUE TITLE
Stack Overflow when 'pub build'

### ISSUE THREAD

**Issue by sestegra** [dart-lang/angular_components_example#23]
# Environment
- Ubuntu 16.04.2 LTS
- Dart SDK 1.23.0
- Below content of .packages

# Scenario
```
$ pub get
$ pub build
Loading source assets... 
Loading angular2/transform/codegen and dart_to_js_script_rewriter transformers... 
Loading angular2 transformers... (4.4s)
Building angular_components_example... (5.6s)
[Info from Dart2JS]:
Compiling angular_components_example|web/main.dart...
[Error from Dart2JS]:
web/main.dart:
The compiler crashed when compiling this element.
The compiler is broken.

When compiling the above element, the compiler crashed. It is not
possible to tell if this is caused by a problem in your program or
not. Regardless, the compiler should not crash.

The Dart team would greatly appreciate if you would take a moment to
report this problem at http://dartbug.com/new.

Please include the following information:

* the name and version of your operating system,

* the Dart SDK build number (build number could not be determined), and

* the entire message you see here (including the full stack trace
  below as well as the source location above).

Build error:
Transform Dart2JS on angular_components_example|web/main.dart threw error: Stack Overflow
package:compiler_unsupported/src/compiler.dart 1479       CompilerDiagnosticReporter.onError
package:compiler_unsupported/src/compiler.dart 311        Compiler.run.<fn>.<fn>
dart:async                                                _Completer.completeError
package:compiler_unsupported/src/compiler.dart 472        Compiler.runInternal
dart:async                                                _rootRunUnary
.                                                         ...
.                                                         ...
dart:async                                                _completeOnAsyncReturn
package:compiler_unsupported/src/library_loader.dart 779  _LibraryLoaderTask.patchLibraryIfNecessary

Build failed.
```

Content of .packages
```
# Generated by pub on 2017-05-05 12:55:25.289640.
analyzer:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/analyzer-0.29.10/lib/
angular2:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/angular2-3.0.0/lib/
angular_components:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/angular_components-0.5.1/lib/
archive:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/archive-1.0.28/lib/
args:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/args-0.13.7/lib/
async:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/async-1.13.2/lib/
barback:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/barback-0.15.2+10/lib/
browser:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/browser-0.10.0+2/lib/
build:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/build-0.8.0/lib/
charcode:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/charcode-1.1.1/lib/
cli_util:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/cli_util-0.0.1+2/lib/
collection:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/collection-1.14.0/lib/
convert:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/convert-2.0.1/lib/
crypto:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/crypto-2.0.1/lib/
csslib:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/csslib-0.13.5/lib/
dart_style:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/dart_style-1.0.4/lib/
dart_to_js_script_rewriter:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/dart_to_js_script_rewriter-1.0.2/lib/
fixnum:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/fixnum-0.10.5/lib/
func:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/func-1.0.0/lib/
glob:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/glob-1.1.3/lib/
html:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/html-0.13.1/lib/
intl:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/intl-0.15.0/lib/
isolate:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/isolate-1.0.0/lib/
js:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/js-0.6.1/lib/
logging:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/logging-0.11.3+1/lib/
matcher:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/matcher-0.12.0+2/lib/
meta:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/meta-1.0.5/lib/
observable:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/observable-0.20.4+1/lib/
package_config:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/package_config-1.0.0/lib/
path:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/path-1.4.1/lib/
perf_api:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/perf_api-0.1.0/lib/
plugin:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/plugin-0.2.0/lib/
pool:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/pool-1.3.0/lib/
protobuf:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/protobuf-0.5.4/lib/
quiver:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/quiver-0.25.0/lib/
source_span:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/source_span-1.3.1/lib/
stack_trace:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/stack_trace-1.7.3/lib/
string_scanner:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/string_scanner-1.0.1/lib/
typed_data:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/typed_data-1.1.3/lib/
unittest:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/unittest-0.11.7/lib/
utf:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/utf-0.9.0+3/lib/
uuid:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/uuid-0.5.3/lib/
watcher:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/watcher-0.9.7+3/lib/
webdriver:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/webdriver-1.2.3/lib/
when:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/when-0.2.0/lib/
which:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/which-0.1.3/lib/
yaml:file:///home/xxx/.pub-cache/hosted/pub.dartlang.org/yaml-2.1.12/lib/
angular_components_example:lib/
```

**Comment by nshahan** [comment]
I noticed this as well. I can run pub serve though. 

This issue has been resolved in the latest version of the SDK. If you upgrade to the 1.24.0-dev.3.0 you should be able to run `pub build`.

**Comment by sestegra** [comment]
The fix should be backported on Dart SDK 1.23.1.

**Comment by nshahan** [comment]
If you want to make a case to have this backported, could you open an issue in the SDK? I believe the fix was related to these commits but @rakudrama would know for sure.

https://github.com/dart-lang/sdk/commit/887e7332036422437a8657aa3478b88091354f8b
https://github.com/dart-lang/sdk/commit/f36aaa72a80b9ee49070a32d45433f6fccd1d405
